src/rust/src/sapling.rs

```diff
@@ -174,3 +331,41 @@ impl Verifier {
             .final_check(value_balance, sighash_value, binding_sig)
     }
 }
+
+struct BatchValidator(Option<sapling_proofs::BatchValidator>);
+
+fn init_batch_validator() -> Box<BatchValidator> {
+    Box::new(BatchValidator(Some(sapling_proofs::BatchValidator::new())))
+}
+
+impl BatchValidator {
+    /// Checks the bundle against Sapling-specific consensus rules, and queues its
+    /// authorization for validation.
+    ///
+    /// Returns `false` if the bundle doesn't satisfy the checked consensus rules. This
+    /// `BatchValidator` can continue to be used, but it may include some or all of the
+    /// bundle's proofs and signatures.
```

Comment:
I don't understand this. What does it mean for a `BatchValidator` to "include" some of the proofs and signatures only when the return value is false? Or is this independent of the return value?